src/main/java/org/opentripplanner/middleware/controllers/api/MonitoredTripController.java

```diff
@@ -161,14 +162,12 @@ private void verifyBelowMaxNumTrips(String userId, Request request) {
      * {@link org.opentripplanner.middleware.otp.response.Itinerary} has transit and no rentals/ride hailing).
      */
     private void checkTripCanBeMonitored(MonitoredTrip trip, Request request) {
-        boolean hasTransit = trip.itinerary.hasTransit();
-        boolean hasRentalOrRideHail = trip.itinerary.hasRentalOrRideHail();
-
-        if (!hasTransit || hasRentalOrRideHail) {
+        Itinerary.ItineraryCanBeMonitored canBeMonitored = trip.itinerary.assessCanBeMonitored();
+        if (!canBeMonitored.overall) {
             String rejectReason = "";
-            if (!hasTransit) rejectReason += "it does not include a transit leg";
-            if (!hasTransit && hasRentalOrRideHail) rejectReason += ", and ";
-            if (hasRentalOrRideHail) rejectReason += "it includes a rental or ride hail";
+            if (!canBeMonitored.hasTransit) rejectReason += "it does not include a transit leg";
+            if (!canBeMonitored.overall) rejectReason += ", and ";
```

Comment:
This doesn't seem future-proofed in case another rejection reason is added later. It might make sense to make this a List of checks that are then joined together with the ", and " string.

Reply:
Addressed, with unit tests for the message, in https://github.com/ibi-group/otp-middleware/pull/109/commits/e85ca83ef2591b8f2834fc3e46a34c3f395803e5.